app/services/foreman_chef/fact_parser.rb

```diff
@@ -36,8 +36,7 @@ def operatingsystem
             minor = release[2..-1]
           end
         when 'centos'
-          # Centos Stream doesn't have minor version on need to replace blank spaces due to name restriction
-          os_name = facts.dig(:os_release, :name).tr(' ', '_') unless minor
+          os_name = 'CentOS'
```

Comment:
Note that previously CentOS 7 was not normalized. This is IMHO also an unrelated fix.